content/spatial/transform.Rmd

```diff
@@ -0,0 +1,54 @@
+---
+title: "Transforming spatial objects"
+date: 2018-02-19T14:52:22+01:00
+---
+
+```{r include = FALSE}
+library(knitr)
+opts_chunk$set(
+  echo = TRUE
+)
+```
+
+
+Sometimes we have a layer in one coordinate system and need to transform it into another coordinate system. The first thing we need to do is identifying both coordinate systems. Let's create an example and identify the coordinates system with `proj4string()`. We used the coordinates posted on the contact page of [NGI](http://www.ngi.be/NL/NL5.shtm).
+
+```{r}
+library(sp)
+library(leaflet)
+```
+
+```{r}
+ngi <- data.frame(x = 650381.78, y = 667603.12)
+coordinates(ngi) <- ~x + y
+proj4string(ngi)
+```
+
+`NA` indicates that the coordinate systems isn't set. So we need to set it manually. In this case we know it is "Lambert 2008". We need to know the related projection string. The projection string is often a rather long string of parameters. However, more projections have an [EPSG](https://en.wikipedia.org/wiki/International_Association_of_Oil_%26_Gas_Producers#European_Petroleum_Survey_Group) number which you can find at http://epsg.io/. The EPSG number for "Lambert 2008" is 3812. Let's set this projection to our dataset. `CRS()` defines the projection based on a text string.
+
+```{r}
+proj4string(ngi) <- CRS("+init=epsg:3812")
+proj4string(ngi)
+```
+
+We could verify the correctness of this position by plotting in on a map. Here we use the `leaflet` package which requires the data to be in the "WGSS84" coordinates system. Therefore we use `spTransform` to do this transformation. "WGS84" has EPSG number 4326. But here the projection string itself is easier to memorise: "+proj=longlat".
```

Comment:
I don't think `+proj=longlat` is advisable to refer to WGS 84; see PR #47

Reply:
What to do with [this](https://github.com/inbo/tutorials/pull/43/files/e413cca05a8e7d0dbf0ab34e3ce6e7ba29459cae#r170956139)? See second bullet [here](https://github.com/inbo/tutorials/pull/47#issue-171691844) for background.
@ThierryO @stijnvanhoey

Reply:
I missed that comment, my excuses. My recommendation is to use the epsg codes (`epsg:4326` for WGS84) as this will be easily reusable in most softwares/langyages/tools. @florisvdh can you do a buf-fix PR?

Reply:
`+proj=longlat` is shorthand for `+proj=longlat +ellps=WGS84`
```
> CRS("+proj=longlat")
CRS arguments: +proj=longlat +ellps=WGS84
```

Reply:
@ThierryO @stijnvanhoey  I understand that the `+proj=longlat` shorthand works; the WGS84 datum is the (silent) default. Still, I prefer being more unequivocal when using a PROJ.4 string, and therefore only use shortcuts (like the `epsg:key` format) that refer to a full PROJ.4 string and aren't a part of one. Many other CRSes exist that use `+proj=longlat`. `+proj=longlat` in itself just means that the CRS is geodetic.

That is my opinion; the decision is up to you guys.

By the way, we must acknowledge that full PROJ.4 strings are not explicit about everything. They are not an OGC standard (while WKT is). WKT is explicit about everything. But PROJ.4
(still) is the workhorse for many applications.

Reply:
I do agree with you, `longlat` is an ambiguous default, so use shortcuts (like the epsg:key format) that refer to a full PROJ.4 string, so +1 for that

And indeed, WKT is more explicit, but epsg-codes and their related PROJ.4 strings are a pragmatic, largely accepted CRS description. (cfr. geojson has also been grown outside OGC, as a pragmatic alternative). 

Are we ok to use (and also 'teach') epsg codes as the *pragmatic* default?

Reply:
Is OK for me.